commit b4e12dec606dc19fa1ce2ba4be39e26e6145cacf
Author: James Antill <james@and.org>
Date:   Thu Jun 23 13:28:23 2011 -0400

    Don't show -c warning, when we've changed it due to --installroot. BZ 707358.

diff --git a/cli.py b/cli.py
index 82f286f..6056d38 100644
--- a/cli.py
+++ b/cli.py
@@ -264,7 +264,9 @@ class YumBaseCli(yum.YumBase, output.YumOutput):
             opts.debuglevel = opts.errorlevel = 6
         if opts.debuglevel != pc.debuglevel or opts.errorlevel != pc.errorlevel:
             self.logger.warning("Ignored option -q, -v, -d or -e (probably due to merging: -yq != -y -q)")
-        if opts.conffile != pc.fn:
+        #  getRoot() changes it, but then setupYumConfig() changes it back. So
+        # don't test for this, if we are using --installroot.
+        if root == '/' and opts.conffile != pc.fn:
             self.logger.warning("Ignored option -c (probably due to merging -yc != -y -c)")
 
         if opts.version:
